### PR TITLE
Actions - Env Vars Improvements

### PR DESCRIPTION
**Pull request by ljtill**
- Refactor workflow

### CODE REVIEW

File: .github/workflows/azops.yml [hunk 1]
```diff
@@ -3,88 +3,90 @@ name: AzOps
...
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  # If you want review before triggering the push event, enable push_with_review block and disable push.
-  # Review will require, someone other than the creator or the PR to complete review.
-  # GitHub Action will run once PR is reviewed with comment that contains  "LGTM". 
-  # However, in order for GitHub Action status check to display in your PR, you have to enable branch access policy.
-  #
-  # push_with_review:
-  #   if: ${{ github.head_ref != 'system' && contains(github.event.review.body, 'LGTM')}}
+
+  # If you want review before triggering the push event, set push_with_review env to true.
```
Comment (krowlandson) [inline .github/workflows/azops.yml:62]: To reduce the need for users to modify this file to make updates from our repo easier, could we leave this uncommented, and use a GitHub Secret to control this section?

i.e. `if: github.head_ref != 'system' && contains(github.event.review.body, 'LGTM') && secrets.push_with_review == 'true'`

If that won't work, would it be useful to include a default entry for "push_with_review: false" in the env block above so it's easier for a user to set?

File: .github/workflows/azops.yml [hunk 2]
```diff
@@ -3,88 +3,90 @@ name: AzOps
...
+      - "submitted"
     paths:
       - "azops/**"
 
+env:
+  AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+  AZOPS_STATE: "azops"
+  AZOPS_DEFAULT_DEPLOYMENT_REGION: "northeurope"
+  AZOPS_ENROLLMENT_ACCOUNT: ""
+  AZOPS_OFFER_TYPE: "MS-AZR-0017P"
```
Comment (krowlandson) [inline .github/workflows/azops.yml:28]: Not really relevant to this PR, but this got me thinking... do we plan to support provisioning "Enterprise Dev/Test" (Offer MS-AZR-0148P) Subscriptions? If so, we would need to add some logic to allow this in AzOps.

File: .github/workflows/azops.yml [hunk 3]
```diff
@@ -3,88 +3,90 @@ name: AzOps
...
     runs-on: ubuntu-latest
     steps:
       - name: Actions checkout
         uses: actions/checkout@v2
       - name: GitHub context
-        env:
-          GITHUB_CONTEXT: ${{ toJson(github) }}
         run: echo "$GITHUB_CONTEXT"
       - name: Actions azops
         uses: Azure/AzOps@main
```
Comment (krowlandson) [inline .github/workflows/azops.yml:58]: Is the intent to always use the original `Azure/AzOps` repository, or should we provide the option to set this via a GitHub secret (with a default of `Azure/AzOps@main` when no secret value provided) so users can override this with their own copy / specific release version if wanted?

Just thinking about this in the context of our discussion yesterday on the team call when we were discussing pinning to specific releases.

Comment (ljtill) [reply .github/workflows/azops.yml:58]: This went through my mind last night; I've not included yet, need to explore options, let's aim to add this in a subsequent pull request.

File: .github/workflows/azops.yml (hunk 2 above)
Comment (ljtill) [reply .github/workflows/azops.yml:28]: Let's take this offline and talk through it.

**Pull request merged**